core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/metrics/TableRepairMetricsImpl.java

```diff
@@ -131,13 +133,24 @@ public Builder withReportInterval(long reportInterval, TimeUnit timeUnit)
             return this;
         }
 
+        public Builder withMetricRegistry(MetricRegistry metricRegistry)
+        {
+            myMetricRegistry = metricRegistry;
+            return this;
+        }
+
         public TableRepairMetricsImpl build()
         {
             if (myTableStorageStates == null)
             {
                 throw new IllegalArgumentException("Table storage states cannot be null");
             }
 
+            if (myMetricRegistry == null)
```

Comment:
Nit-picking:

Recently we have started using a pattern like the following in the constructors:
`
myMetricRegistry = Preconditions.checkNotNull(builder.myMetricRegistry, "Metric registry cannot be null");
`
Which IMO is cleaner than if and manual throws.